salt/modules/debian_ip.py

```diff
@@ -1498,6 +1498,13 @@ def _write_file_routes(iface, data, folder, pattern):
     '''
     Writes a file to disk
     '''
+    # Since the entire directory is executed with matching pattern
+    # ^[a-zA-Z0-9_-]+$ not all filenames are legitimate.
+    # In order to make routes work for vlan interfaces (e.g. bond0.12)
+    # the dots in the iface name will be replaced by underscores, so it
+    # will be properly executed
```

Comment:
From what i can find online, ifup / ifdown probably uses run-parts to execute a directory.

From man-page run-parts:
>
> run-parts runs all the executable files named within constraints described below, found in 
> directory directory.  Other files and directories are silently ignored.
>
> If neither the --lsbsysinit option nor the --regex option is given then the names must consist
> entirely of ASCII upper- and lower-case letters, ASCII digits, ASCII underscores, and ASCII minus-hyphens.
>
> (^[a-zA-Z0-9_-]+$)